packages/data-service/src/data-service.ts

```diff
@@ -2314,25 +2311,21 @@ class DataServiceImpl extends WithLogContext implements DataService {
     return { name, ...normalized };
   }
 
-  @op(mongoLogId(1_001_000_063))
+  @op(mongoLogId(1_001_000_266))
   async previewUpdate(
     ns: string,
     filter: Document,
     update: Document | Document[],
-    executionOptions?: UpdatePreviewExecutionOptions
+    executionOptions: UpdatePreviewExecutionOptions = {}
   ): Promise<UpdatePreview> {
-    if (!executionOptions) {
-      const controller = new AbortController();
-      executionOptions = {
-        abortSignal: controller.signal,
-        sample: PREVIEW_SAMPLE,
-        timeout: PREVIEW_TIMEOUT,
-      };
-    }
-
-    const sample = executionOptions.sample || PREVIEW_SAMPLE;
-    const timeout = executionOptions.timeout || PREVIEW_TIMEOUT;
-
+    const {
+      abortSignal = new AbortController().signal,
```

Comment:
(previously, passing `executionOptions` with `sample` and `timeout` would have errored because no session is created if `abortSignal` is nullish)